src/Proxy/DriveItemProxy.php

```diff
@@ -295,6 +296,54 @@ public function upload($name, $content, array $options = [])
 
         return new self($this->graph, $driveItem);
     }
+	
+	/**
+     * Creates a upload session to upload a file under this folder drive item.
+     *
+     * @param string $name
+     *        The name.
+     * @param string|resource|\GuzzleHttp\Psr7\Stream $content
+     *        The content.
+     * @param array $options
+     *        The options.
+     *
+     * @return UploadSessionProxy
+     *         The upload session created.
+     *
+     * @todo Support content type in options.
+     */
+    public function startUpload($name, $content, array $options = [])
+    {
+		$name         = rawurlencode($name);
+        $driveLocator = "/drives/{$this->parentReference->driveId}";
+        $itemLocator  = "/items/{$this->id}";
+		$endpoint = "$driveLocator$itemLocator:/$name:/createUploadSession";
+		
+		//Include extended options with existing item for upload session. Extended options to be overwritten if not mentioned
+		$item = [
+			'name' => isset($options['name']) ? $options['name'] : $name,
+			'description' => isset($options['name']) ? $options['description'] : '',
+			'@microsoft.graph.conflictBehavior' => isset($options['conflictBehavior']) ? $options['conflictBehavior'] : 'rename',
+		];
+		
+		//create an upload session for large files
+		$response = $this
+			->graph
+			->createRequest('POST', $endpoint)
+			->addHeaders(array('Content-Type' => 'application/json'))
```

Comment:
I looked into other API calls, we do not explicitly set the request content type to `application/json`: this is probably because it is done internally by the Graph library. We should remove it for consistency.

Reply:
I interpreted the request directly from Microsoft Graph explorer online. Hence unaware of the internals of the request. Ok I'll have it removed.

Reply:
@krizalys I had a check in Microsoft Graph Explorer just now. When I remove this from header, it interprets the body as text/plain giving me the error response as,

`{
    "error": {
        "code": "",
        "message": "The request entity's media type 'text/plain' is not supported for this resource.",
        "innerError": {
            "request-id": "1d11c78b-4377-4634-b395-16a416415552",
            "date": "2019-02-21T09:19:15"
        }
    }
}`

Are you sure you want me to remove this ?